coderd/httpapi/httpapi.go

```diff
@@ -102,10 +106,13 @@ func Write(rw http.ResponseWriter, status int, response interface{}) {
 
 // Read decodes JSON from the HTTP request into the value provided.
 // It uses go-validator to validate the incoming request body.
-func Read(rw http.ResponseWriter, r *http.Request, value interface{}) bool {
+func Read(ctx context.Context, rw http.ResponseWriter, r *http.Request, value interface{}) bool {
+	ctx, span := tracing.StartSpan(ctx)
+	defer span.End()
```

Comment:
Why do we care about tracing this? It seems like this function is so simple, that it's weird to trace.

Reply:
For `httpapi.Read` I would agree, but that was mostly to make them consistent. I thought it would be weird if one was traced and the other wasn't.

`httpapi.Write` is a lot more useful. JSON marshaling can sometimes take a non-insignificant amount of time, and this could help us catch outliers. Additionally, we can enrich span data a bit more since we have access to the actual `interface{}` we're marshalling, such as the number of elements in an array, which could help us spot routes that need to be paginated.

Reply:
Toss this gorgeous comment next to `http.Write` and I'll approve!